src/main/java/paul/storage/Storage.java

```diff
@@ -0,0 +1,114 @@
+package paul.storage;
+
+import paul.exception.PaulException;
+import paul.task.Deadline;
+import paul.task.Event;
+import paul.task.Task;
+import paul.task.TaskList;
+import paul.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * Handles reading from and writing to the storage file,
+ * which keeps track of tasks.
+ */
+public class Storage {
+    // Path to the storage file
+    private final String filePath;
+
+    /**
+     * Creates a Storage instance with the given file path.
+     * Creates the directory if it does not exist.
+     *
+     * @param filePath The path to the storage file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+
+        // create directory if it does not exist
+        File parentFile = new File(filePath).getParentFile();
+        if (!parentFile.exists()) {
+            if (!parentFile.mkdirs()) {
+                System.out.println("Error: Could not create folder.");
+            }
+        }
+    }
+
+    /**
+     * Loads tasks from the file into a TaskList.
+     *
+     * @return Tasklist containing the tasks from the storage.
+     * @throws PaulException if the file is not found.
+     */
+    public TaskList loadTasks() throws PaulException {
+        TaskList tasks = new TaskList();
+        File file = new File(filePath);
+
+        if (!file.exists()) {
+            return tasks;
+        }
+
+        try (Scanner sc = new Scanner(file)) {
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine().trim();
+                String[] parts = line.split(" \\| ");
+                String type = parts[0]; // T, D, E
+                boolean isDone = parts[1].equals("1");
+                String description = parts[2];
+
+                Task task;
+                switch (type) {
+                    case "T" -> task = new ToDo(description);
+                    case "D" -> task = new Deadline(description, LocalDate.parse(parts[3]));
+                    case "E" -> task = new Event(
+                            description, LocalDate.parse(parts[3]), LocalDate.parse(parts[4]));
+                    default -> {
+                        System.out.println("Unknown task type in file, skipping: " + type);
+                        continue;
+                    }
+                }
```

Comment:
Would it be better to standardise the switch statements? This would make it easier to follow.

```suggestion
                switch (type) {
                    case "T":
                        task = new ToDo(description);
                        break;
                    case "D":
                        task = new Deadline(description, LocalDate.parse(parts[3]));
                        break;
                    case "E":
                        task = new Event(
                        description, LocalDate.parse(parts[3]), LocalDate.parse(parts[4]));
                        break;
                    default:
                        System.out.println("Unknown task type in file, skipping: " + type);
                        continue;
                }

```